driver/gl/window.go

```diff
@@ -503,50 +593,104 @@ func keyToName(key glfw.Key) fyne.KeyName {
 		return desktop.KeySuper
 	case glfw.KeyMenu:
 		return desktop.KeyMenu
-
-	case glfw.KeyKPEnter:
-		return fyne.KeyEnter
 	}
 	return ""
 }
 
 func (w *window) keyPressed(viewport *glfw.Window, key glfw.Key, scancode int, action glfw.Action, mods glfw.ModifierKey) {
-	if w.canvas.Focused() == nil && w.canvas.onTypedKey == nil {
-		return
-	}
+
 	if action != glfw.Press { // ignore key up
+		// TODO: handle repeated key
 		return
 	}
 
-	if key <= glfw.KeyWorld1 { // filter printable characters handled in charModInput
+	keyName := keyToName(key)
+	keyDesktopModifier := desktopModifier(mods)
+
+	if keyDesktopModifier < desktop.ControlModifier {
+		// No shortcut detected, pass down to TypedKey and exit
+		keyEvent := &fyne.KeyEvent{Name: keyName}
+		if w.canvas.Focused() != nil {
+			go w.canvas.Focused().TypedKey(keyEvent)
+		}
+		if w.canvas.onTypedKey != nil {
+			go w.canvas.onTypedKey(keyEvent)
+		}
 		return
 	}
 
-	ev := new(fyne.KeyEvent)
-	ev.Name = keyToName(key)
+	isMac := runtime.GOOS == "darwin"
 
-	if w.canvas.Focused() != nil {
-		go w.canvas.Focused().TypedKey(ev)
+	var shortcut fyne.Shortcuter
```

Comment:
I think the logic in this section is inside out...

if the "if control || command" was on the outside and the switch for shortcut key was on the inside it would be much cleaner.